src/unix/solarish/mod.rs

```diff
@@ -489,25 +510,35 @@ s! {
 s_no_extra_traits! {
     #[cfg_attr(any(
         target_arch = "x86", target_arch = "x86_64"),
-        repr(packed(4))
+        repr(packed(8))
     )]
-    pub struct epoll_event {
-        pub events: u32,
-        pub u64: u64,
-    }
-
     pub struct utmpx {
+        #[cfg(target_os = "illumos")]
         pub ut_user: [::c_char; _UTX_USERSIZE],
+        #[cfg(target_os = "solaris")]
+        pub ut_user: [::c_char; _UTMP_USER_LEN],
+        #[cfg(target_os = "illumos")]
         pub ut_id: [::c_char; _UTX_IDSIZE],
+        #[cfg(target_os = "solaris")]
+        pub ut_id: [::c_char; _UTMP_ID_LEN],
+        #[cfg(target_os = "illumos")]
         pub ut_line: [::c_char; _UTX_LINESIZE],
+        #[cfg(target_os = "solaris")]
+        pub ut_line: [::c_char; _UTMP_LINE_LEN],
         pub ut_pid: ::pid_t,
         pub ut_type: ::c_short,
         pub ut_exit: exit_status,
         pub ut_tv: ::timeval,
         pub ut_session: ::c_int,
+        #[cfg(target_os = "illumos")]
         pub ut_pad: [::c_int; _UTX_PADSIZE],
+        #[cfg(target_os = "solaris")]
+        pub pad: [::c_int; 5],
         pub ut_syslen: ::c_short,
+        #[cfg(target_os = "illumos")]
         pub ut_host: [::c_char; _UTX_HOSTSIZE],
+        #[cfg(target_os = "solaris")]
+        pub ut_host: [::c_char; 257],
```

Comment:
Rather than this `cfg()` soup (here and for some of the other modified structs), it might make more sense to move to separate definitions in `illumos.rs` and `solaris.rs`